.github/workflows/cpp.yml

```diff
@@ -152,20 +152,20 @@ jobs:
       ARROW_WITH_BROTLI: OFF
       ARROW_USE_GLOG: OFF
       ARROW_BUILD_TESTS: ON
-      ARROW_TEST_LINKAGE: static
+      ARROW_BUILD_SHARED: ON
+      ARROW_BUILD_STATIC: OFF
       ARROW_BOOST_USE_SHARED: OFF
-      ARROW_BUILD_SHARED: OFF
-      ARROW_USE_STATIC_CRT: ON
       ARROW_VERBOSE_THIRDPARTY_BUILD: OFF
     steps:
+      - name: Disable Crash Dialogs
+        run: reg add "HKCU\SOFTWARE\Microsoft\Windows\Windows Error Reporting" /v DontShowUI /t REG_DWORD /d 1 /f
```

Comment:
@kszucs Do you want me to put this in a script?

Reply:
This is quite environment specific so this is the right place.

Reply:
This can actually be useful on all Windows builders.

Reply:
Note this is not required for this PR, but was part of my investigations and save future hassles.

Reply:
@pitrou then what was the solution? The hanging was just transient?